modules/signals/spec/signal-state.spec.ts

```diff
@@ -194,4 +194,28 @@ describe('signalState', () => {
       expect(stateCounter).toBe(3);
       expect(userCounter).toBe(1);
     }));
+
+  describe('freezeInDevMode', () => {
+    it('throws on a mutable change', () => {
+      const userState = signalState(initialState);
+      expect(() =>
+        patchState(userState, (state) => {
+          state.ngrx = 'mutable change';
+          return state;
+        })
+      ).toThrowError("Cannot assign to read only property 'ngrx' of object");
+    });
+
+    it('throws on a mutable change', () => {
+      const userState = signalState(initialState);
+      expect(() =>
+        patchState(userState, (state) => {
+          state.user.firstName = 'mutable change';
+          return state;
+        })
+      ).toThrowError(
+        "Cannot assign to read only property 'firstName' of object"
+      );
+    });
+  });
```

Comment:
I suggest moving these tests to `deep-freeze.spec.ts` and checking the behavior for both - `signalState` and `signalStore`. Take a look at these tests for inspiration: https://github.com/ngrx/platform/blob/main/modules/signals/spec/state-source.spec.ts#L49

Also, in the following cases, error should be thrown as well:

```ts
userState.user().firstName =  'mutable change 1'; // error

// ---

getState(userState).ngrx = 'mutable change 2'; // error

// ---

const s = { user: { firstName: 'M', lastName: 'S' } };
patchState(userState, s);
s.user.firstName = 'mutable change 3'; // error
```